src/engine/LoadURL.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 The QLever Authors, in particular:
+//
+// 2025 Julian Mundhahs <mundhahj@tf.uni-freiburg.de>, UFR
+//
+// UFR = University of Freiburg, Chair of Algorithms and Data Structures
+
+#include "engine/LoadURL.h"
+
+#include "global/RuntimeParameters.h"
+#include "util/http/HttpUtils.h"
+
+// _____________________________________________________________________________
+string LoadURL::getCacheKeyImpl() const {
+  // TODO<qup42> do caching based on ETag, Last-Modified or similar
+  if (RuntimeParameters().get<"cache-load-results">()) {
+    return absl::StrCat("LOAD URL ", loadURLClause_.url_.asString(),
+                        loadURLClause_.silent_ ? " SILENT" : "");
+  }
+  return absl::StrCat("LOAD URL ", cacheBreaker_);
+}
+
+// _____________________________________________________________________________
+string LoadURL::getDescriptor() const {
+  return absl::StrCat("LOAD URL ", loadURLClause_.url_.asString());
+}
+
+// _____________________________________________________________________________
+size_t LoadURL::getResultWidth() const { return 3; }
+
+// _____________________________________________________________________________
+size_t LoadURL::getCostEstimate() {
+  // TODO: For now, we don't have any information about the cost at query
+  // planning time, so we just return ten times the estimated size.
+  return 10 * getSizeEstimateBeforeLimit();
+}
+
+// _____________________________________________________________________________
+uint64_t LoadURL::getSizeEstimateBeforeLimit() {
+  // TODO: For now, we don't have any information about the result size at
+  // query planning time, so we just return `100'000`.
```

Comment:
Is it also true, that this will always be the only operation in a tree in reall life, so the estimates don't matter because there is nothing to plan?
Then this could also be stated here.